GitCommands/Git/GitDirectoryResolver.cs

```diff
@@ -15,12 +16,14 @@ public interface IGitDirectoryResolver
         /// </summary>
         /// <param name="repositoryPath">The repository working folder.</param>
         /// <returns>The resolved location of .git folder.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="repositoryPath"/> is <see langword="null"/>.</exception>
         string Resolve(string repositoryPath);
     }
 
     /// <summary>
     /// Resolves the location of .git folder.
     /// </summary>
+    [Export(typeof(IGitDirectoryResolver))]
```

Comment:
💡 I recommend making the following changes:

1. Eliminate all but one constructor
2. Mark the remaining constructor with `[ImportingConstructor]`

Reply:
Can you please help me understanding how to register `FileSystem -> IFileSystem`?

```diff
diff --git a/GitCommands/Git/GitDirectoryResolver.cs b/GitCommands/Git/GitDirectoryResolver.cs
index 1d81f491f..d3610c9fc 100644
--- a/GitCommands/Git/GitDirectoryResolver.cs
+++ b/GitCommands/Git/GitDirectoryResolver.cs
@@ -28,16 +28,12 @@ public sealed class GitDirectoryResolver : IGitDirectoryResolver
     {
         private readonly IFileSystem _fileSystem;
 
+        [ImportingConstructor]
         public GitDirectoryResolver(IFileSystem fileSystem)
         {
             _fileSystem = fileSystem;
         }
 
-        public GitDirectoryResolver()
-            : this(new FileSystem())
-        {
-        }
-
```